src/Consolonia.Core/Infrastructure/ConsoleCursorFactory.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+using System.Collections.Generic;
+using Avalonia;
+using Avalonia.Input;
+using Avalonia.Platform;
+
+namespace Consolonia.Core.Infrastructure
+{
+#pragma warning disable CA1711 // Identifiers should not have incorrect suffix
+    internal sealed class ConsoleCursorFactory : ICursorFactory
+    {
+        private readonly Dictionary<StandardCursorType, CursorImpl> _standardCursors = new();
+
+        public ICursorImpl GetCursor(StandardCursorType cursorType)
```

Comment:
Just in case you don't know. you can simply mark method with [MethodImpl(Synchronized)]

Reply:
What does that do?

Reply:
it locks by the current instance which contains this method